### PR TITLE
#619 Implemented SumOfScalar to sum up values from its scalars

### PR DESCRIPTION
**Pull request by nick318**
#619 Implemented SumOfScalar to sum up values from its scalars

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#651](https://github.com/yegor256/cactoos/issues/651) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#651](https://github.com/yegor256/cactoos/issues/651) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/C63314D6Z/461213462). The budget is 15 minutes, see [§4](http://datum.zerocracy.com/pages/policy.html#4). Please, read [§27](http://datum.zerocracy.com/pages/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket. If you decide that this PR should not be accepted ever, also inform the architect.

**Comment by amihaiemil** [comment]
@nick318 As I know, we already have class ``SumOf`` -- can't that be used somehow, maybe with some lambdas, to sum up Scalars? 

And this class, SumOfScalars, is not named properly, in my opinion, because it implements ``Scalar<Integer>`` -- it won't sum any Scalars, just the Integer ones, right?

**Comment by nick318** [comment]
@amihaiemil That is exactly `SumOf` what I use inside.
About name I agree, would it be better to name `SumOfIntScalars`?

**Comment by amihaiemil** [comment]
@nick318 Yes, SumOfIntScalars sounds better. And could you add one more test, for empty list? Or a list with only one element? How does it behave?

**Comment by nick318** [comment]
@amihaiemil take a look please to refreshed version. I encapsulated logic of summing up scalars to internal class in then made a different classes for each type of number. Also added few tests as you asked for.

**Comment by nick318** [comment]
@amihaiemil ping

File: src/main/java/org/cactoos/scalar/SumOfDoubleScalar.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+ *
+ * <p>There is no thread-safety guarantee.
+ *
+ * @author Nikita Salomatin (nsalomatin@hotmail.com)
+ * @version $Id$
+ * @since 0.30
+ */
+@SuppressWarnings(
+    {
+        "PMD.CallSuperInConstructor",
```
Comment (amihaiemil) [inline src/main/java/org/cactoos/scalar/SumOfDoubleScalar.java]: @nick318 Are all these suppressions needed for all the classes?

File: src/main/java/org/cactoos/scalar/SumOfFloatScalar.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.scalar;
+
+import org.cactoos.Scalar;
+
+/**
+ * Make a scalar which is sum of scalar's values.
```
Comment (amihaiemil) [inline src/main/java/org/cactoos/scalar/SumOfFloatScalar.java]: @nick318 "Float | Double | Integer Scalar which sums up the values of other Scalars of the same type".

File: src/test/java/org/cactoos/scalar/SumOfIntScalarTest.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+/**
+ * Test case for {@link SumOfIntScalar}.
+ *
+ * @author Nikita Salomatin (nsalomatin@hotmail.com)
+ * @version $Id$
+ * @since 0.30
+ * @checkstyle JavadocMethodCheck (500 lines)
+ * @checkstyle MagicNumberCheck (500 lines)
+ */
+@SuppressWarnings("PMD.TooManyMethods")
```
Comment (amihaiemil) [inline src/test/java/org/cactoos/scalar/SumOfIntScalarTest.java]: @nick318 They don't seem like a lot of methods, are you sure you need this?

File: src/main/java/org/cactoos/scalar/SumOfDoubleScalar.java (hunk 1 above)
Comment (nick318) [reply src/main/java/org/cactoos/scalar/SumOfDoubleScalar.java]: @amihaiemil no, I just deleted it, thanks.

File: src/main/java/org/cactoos/scalar/SumOfFloatScalar.java (hunk 2 above)
Comment (nick318) [reply src/main/java/org/cactoos/scalar/SumOfFloatScalar.java]: @amihaiemil done

File: src/test/java/org/cactoos/scalar/SumOfIntScalarTest.java (hunk 3 above)
Comment (nick318) [reply src/test/java/org/cactoos/scalar/SumOfIntScalarTest.java]: @amihaiemil Just deleted, thanks

**Comment by nick318** [comment]
@amihaiemil I've done you remarks, take a look please

**Comment by amihaiemil** [comment]
@rultor good to merge

**Comment by nick318** [comment]
@yegor256 take a look please

**Comment by amihaiemil** [comment]
@yegor256 ping

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job, as in [§30](http://datum.zerocracy.com/pages/policy.html#30)

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
> @0crat quality good ([here](https://github.com/yegor256/cactoos/issues/651#issuecomment-366808003))

@ypshenychka Job `gh:yegor256/cactoos#651` is not assigned, can't get start time

**Comment by 0crat** [comment]
QA review completed: +15 points just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka), total is [+210](https://www.0crat.com/u/ypshenychka)

**Comment by 0crat** [comment]
QA review completed: +15 points just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka), total is [+210](https://www.0crat.com/u/ypshenychka)

**Comment by 0crat** [comment]
Order was finished, quality was "good": +20 points just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), total is [+695](https://www.0crat.com/u/amihaiemil)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://datum.zerocracy.com/pages/policy.html#28): +15 points just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256), total is [+10244](https://www.0crat.com/u/yegor256)
